ibis/backends/pandas/execution/window.py

```diff
@@ -183,7 +184,9 @@ def trim_with_timecontext(data, timecontext: Optional[TimeContext]):
     # Filter the data, here we preserve the time index so that when user is
     # computing a single column, the computation and the relevant time
     # indexes are retturned.
-    if TIME_COL not in df:
+    if TIME_COL not in df or not isinstance(
+        df[TIME_COL].dtype, DatetimeTZDtype
```

Comment:
but rather than this use `is_datetime64tz_dtype` (just checks if its a tz-aware dtype)